File: app/models/emox/folder_info.rb
module Emox
  class FolderInfo
    include ApplicationMemoryRecord
    memory_record [
      { key: :active, name: "公開",   icon: "check",             type: "is-primary", },
      { key: :draft,  name: "下書き", icon: "lock-outline",      type: "is-warning", },
      { key: :trash,  name: "ゴミ箱", icon: "trash-can-outline", type: "is-danger",  },
    ]
  end
end

File: nuxt_side/components/Kiwi/models/folder_info.js
import { ApplicationMemoryRecord } from "@/components/models/application_memory_record.js"

export class FolderInfo extends ApplicationMemoryRecord {
  static get define() {
    return [
      { key: "public",   name: "公開",     icon: "eye-outline",  type: null, message: "誰でも見れる", },
      { key: "limited",  name: "限定公開", icon: "link",         type: null, message: "リンクを伝えた人だけ見れる", },
      { key: "private",  name: "非公開",   icon: "lock",         type: null, message: "本人だけ見れる", },
    ]
  }
}
